problem 5.2

ZaicevDima opened the pull request.

ZaicevDima committed:
problem 5.2

ZaicevDima committed:
function sum in decimal number system addaed

jzuken commented on 5дз/2.qbs:
```diff
@@ -0,0 +1,19 @@
+import qbs
```
оно тут в двух экземплярах всё

jzuken commented on 5дз/5.2/main.cpp:
```diff
@@ -0,0 +1,30 @@
+    bool* bynaryNumber2 = binaryTranslation(number2);
+    printBinaryTranslation(bynaryNumber2);
+    cout << "\n";
+    bool*  bynarySum= sum(bynaryNumber1, bynaryNumber2);
```
пробел

jzuken commented on 5дз/5.2/main.cpp:
```diff
@@ -0,0 +1,30 @@
+    int number1 = 0;
+    int number2 = 0;
+    cin >> number1 >> number2;
+    bool* bynaryNumber1 = binaryTranslation(number1);
```
это слово тоже стоит посмотреть в словаре и переименовать тут всё

jzuken commented on 5дз/bynary.cpp:
```diff
@@ -0,0 +1,69 @@
+
+    if (carry)
+    {
+        bool *overcoming = new bool[maxLength];
```
а это кто удалит? проверь тут работу с памятью, пожалуйста

jzuken commented on 5дз/bynary.cpp:
```diff
@@ -0,0 +1,69 @@
+{
+    bool *number = new bool[maxLength];
+    fill(number, number + maxLength, 0);
+    bool carry = false;
```
булевы значения обычно именуют, начиная с префикса типа is

jzuken commented on 5дз/bynary.cpp:
```diff
@@ -0,0 +1,69 @@
+    return bytes;
+}
+
+bool* sum(bool *number1,bool *number2)
```
пробел

ZaicevDima committed:
added spaces, cleared memory, changed type

jzuken commented on first semester/hw5/5.2/binary.cpp:
```diff
@@ -28,52 +28,55 @@ bool* binaryTranslation(int number)
         fill(overcoming, overcoming + maxLength, 0);
         overcoming[0] = 1;
         sum(number, overcoming);
+        delete [] overcoming;
```
это не всё

ZaicevDima committed:
deleted deleted numbers, bytes, added space

ZaicevDima committed:
changed print sum in decimal system

ZaicevDima committed:
cange binary.cpp

ZaicevDima committed:
changed delete(again)

ZaicevDima committed:
deleted unnecessary features

ZaicevDima committed:
deleted comment, and file  pro.user

ZaicevDima committed:
deleted double tab

ZaicevDima committed:
Update binary.h

ZaicevDima committed:
deleted functions of .h

ZaicevDima committed:
1
Merge branch 'hw5.2' of https://github.com/ZaicevDima/HW-144.math.spbu into hw5.2

jzuken commented:
тут конфликт с мастером образовался, надо бы его исправить

jzuken commented:
и вообще, зачем тут какие-то исправления отступов?

ZaicevDima committed:
Merge branch 'master' into hw5.2
